docs/plugin/messaging.md

```diff
@@ -0,0 +1,601 @@
+In the past, logging, tracing and warning reporting were completely separate issues and were handled individually. However, even though these message types have distinct characteristics, we recognized the advantages of consolidating them into a unified framework, which is now referred to as `Messaging`. We are convinced it is much better suited for the development, and with some recent fixes that will be described in the following sections, `Messaging` is fully operational. We strongly believe that it is a good time to shed some more light on that functionality, since it is simple to use and yet very effective.
+
+Before we begin to describe what the `MessageControl` plugin does and where logging, tracing and warning reporting can be directed, it is probably a good idea to start from the ground up. In this case, what precisely are these message types, what are their differences in general and in Thunder, how exactly the messages are created, and what is going on with them before they end up at their final destination.
+
+We can all agree that logging, tracing and warning reporting are important techniques often used in software development to gather information and provide insight into the behavior of an application. However, there are some fundamental differences between them.
+
+# Tracing
+
+First, let us briefly discuss the part responsible for the tracing. In theory, tracing is the process of monitoring the flow of a request through an application. It is used to identify performance bottlenecks and understand the interactions between different components of a distributed system. A trace typically consists of a series of events, each of which corresponds to a particular stage in the processing of a request. Tracing provides a much broader and more continuous perspective of the application compared to logging. The goal of tracing is to track the flow and evolution of data within a program so that we can be proactive instead of just reactive and increase overall performance.
+
+```c++
+TRACE(Trace::Information, (_T("Not an A2DP audio sink device!")));
+
+TRACE_GLOBAL(Trace::Error, ("Is this not a descriptor to a DRM Node... =^..^= "));
+```
+
+In Thunder, there are two main macros that can be used to wrap messages that will be treated as traces. These macros, namely `TRACE` and `TRACE_GLOBAL`, can be found in `Thunder/Source/messaging/TraceControl.h`. Upon examining the code, we can notice that the only distinction between them is that `TRACE` includes the class name where it is used, whereas the global version refers to the function name instead of a class.
```

Comment:
Yes, `TRACE` should always be used if we have the `this` pointer available, so when tracing takes place inside a class, otherwise you have to use `TRACE_GLOBAL`. That's a good point though and I'll include that in the documentation, maybe even as a warning.